questionnaire.md

```diff
@@ -0,0 +1,232 @@
+
+    const numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+
+    numbers.forEach((num) => console.log(num)); //In this case is better use forEach because
+                                                //I only want to show the numbers
+
+    const powNumbers = numbers.map((num) => num * num); //In this case is better use map because
+                                                        //I need a new array with changes.
+
+> **What is the reduce use for? Provide examples.**
+>
+> > Reduce is a method that will loop over items in an array and every single time that you loop
+> > over an item in an array, you have the option to return a value and you can use to
+> > accumulate values.
+
+    const numbers = [1, 2, 3, 4];
+    const total = numbers.reduce((total, item) => total + item, 0); // total = 10
+
+> **What is the difference between regular functions and arrow functions? Provide examples.**
+>
+> > The main differences between them are in how they handle the "this" keyword, the
+> > "arguments" object.
+> >
+> > 1. "this" keyword: Arrow functions do not have their own "this" keyword.
+
+    function greetRegular() {
+    console.log('Hello, world!');
+    console.log(this);
+    }
+    greetRegular(); // show: Object [global] {...}...
+
+
+    const greetArrow = () => {
+    console.log('Hello, world!');
+    console.log(this); // "this" refers to the lexical scope where the function is defined
+    };
+
+    greetArrow(); // show an entity object: {}
+
+> > 2.  "arguments" object: Arrow functions do not have their own "arguments" object,
+> >     regular functions have their own "arguments" object which represents the named
+> >     and unnamed arguments passed to the function.
+
+> **What is a higher order function? Provide examples.**
+>
+> > High order functions are methods that pass as arguments another function (callback) or function that return another function.
+
+> **What is the difference between filter and find? Provide examples.**
+>
+> > _find_ return the first element that meets the condition
+> >
+> > _filter_ return a new array with all elements that meets the condition
+
+    const numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+    const oneNumber = numbers.find((num) => num > 5); // return 6
+    const biggerThanFive = numbers.filter((num) => num > 5); // return [ 6, 7, 8, 9 ]
+
+> **What does it mean to pass variables by value? Provide examples.**
+>
+> > Passing variables by value means that you pass a copy of the value of the variable, and if you make a change
+> > inside the function you will no change the original variable. Example:
+
+    let firstName = 'fabio';
+    function showNameUpperCase(name) {
+        name = name.toUpperCase();
+        console.log(name);
+    }
+    console.log(firstName) // fabio
+    showNameUpperCase(firstName);
+    console.log(first) // fabio
```

Comment:
where did you define 'first' in the example

Reply:
Sorry It was a mistake. It was 'firstName'